src/native/minipal/thread.h

```diff
@@ -44,7 +44,10 @@ extern "C" {
 static inline size_t minipal_get_current_thread_id(void)
 {
 #if defined(__wasm) && !defined(_REENTRANT)
-    return 0;
+    pthread_t this_thread = pthread_self();
+    size_t tid = 0;
+    memcpy(&tid, &this_thread, sizeof(this_thread) < sizeof(tid) ? sizeof(this_thread) : sizeof(tid));
+    return tid;
```

Comment:
```suggestion
    return (size_t)(void*)pthread_self();
```
We do this below. Any reason why we can't do this here as well?

Reply:
That isn't really a good practice. It is assuming an implementation detail of `pthread_t`, likely that `typedef void* pthread_t;`, The more robust way is to validate the size of both and copy over the minimum size of both to a `size_t`. I could rework this for them to both be that if you'd like.

Reply:
> For example, it may not work for big endian platforms, or it may not work if pthread_t is a struct with uninitialized padding.

The above operation is defined and correct in all those cases. What we're debating here is does it represent uniqueness in a manner we desire for a thread ID, which is fair. The last blocks of operations are ostensibly the same, but someone needed a cast. We could remove all of these and replace them with the well-defined `memcpy` and be done with it.

> This is not a portable code and that's why we have a battery of ifdefs here. 

Sure. I would prefer to have a fallback, which we're clearly fine with given that is the final block, that is well-defined in the default case and only special case as we discover needed.